app/src/main/java/com/alexstyl/themingandroid/AppCompatThemedActivity.java

```diff
@@ -38,10 +38,10 @@ protected final void applyTheme(AppTheme theme) {
     }
 
     private void restartActivity() {
-        Intent intent = getIntent();
+        final Intent intent = getIntent();
```

Comment:
Sometimes compiler makes appropriate optimizations when it seens the `final` keyword. That's not always guaranteed tho. Either way it wouldn't hurt having it in this case.